std/conv.d

```diff
@@ -875,11 +875,19 @@ T toImpl(T, S)(S value)
         ()@trusted{ memcpy(result.ptr, value.ptr, value.length); }();
         return cast(T) result;
     }
-    else static if (isPointer!S && is(S : const(char)*))
+    else static if (isPointer!S && (is(S : const(char)*) || is(S : const(wchar)*) || is(S : const(dchar)*)))
     {
-        import core.stdc.string : strlen;
-        // It is unsafe because we cannot guarantee that the pointer is null terminated.
-        return value ? cast(T) value[0 .. strlen(value)].dup : null;
+        // This is unsafe because we cannot guarantee that the pointer is null terminated.
```

Comment:
No it doesn't. That's actually what's causing the auto-tester to fail win32 - std.datetime does std.conv.to!string(wchar*). It's obvious the code in std.datetime has not been tested, since to!string(wchar*) did *the wrong thing* until this PR.

That said, it's not actually less safe than what was done before.